pkg/txm/storage/inmemory_store.go

```diff
@@ -269,6 +271,27 @@ func (m *InMemoryStore) UpdateTransactionBroadcast(txID uint64, txNonce uint64,
 	return nil
 }
 
+// Shouldn't call lock because it's being called by a method that already has the lock
+func (m *InMemoryStore) pruneUnstartedTransactionsWithinDuration(threshold time.Duration) []uint64 {
+	var txIDsToPrune []uint64
+	idxTxToRetain := 0
+	for ; idxTxToRetain < len(m.UnstartedTransactions); idxTxToRetain++ {
+		tx := m.UnstartedTransactions[idxTxToRetain]
+		if time.Since(tx.CreatedAt) < threshold {
+			break
+		}
+		txIDsToPrune = append(txIDsToPrune, tx.ID)
+		delete(m.Transactions, tx.ID)
+		m.UnstartedTransactions[idxTxToRetain] = nil // prevent memory leak
+	}
+	if len(txIDsToPrune) == 0 {
+		return nil
+	}
+	m.UnstartedTransactions = m.UnstartedTransactions[idxTxToRetain:]
+	sort.Slice(txIDsToPrune, func(i, j int) bool { return txIDsToPrune[i] < txIDsToPrune[j] })
+	return txIDsToPrune
+}
+
```

Comment:
Does the work 👍 . Here's a more idiomatic way for future reference:
```suggestion
// Shouldn't call lock because it's being called by a method that already has the lock
func (m *InMemoryStore) pruneUnstartedTransactionsWithinDuration(threshold time.Duration) (txIDsToPrune []uint64) {
	for i, tx := range m.UnstartedTransactions {
		if time.Since(tx.CreatedAt) < threshold {
			m.UnstartedTransactions = m.UnstartedTransactions[i:]
			return txIDsToPrune // you can sort before this if you want to
		}
		txIDsToPrune = append(txIDsToPrune, tx.ID)
		delete(m.Transactions, tx.ID)
		m.UnstartedTransactions[i] = nil // prevent memory leak
	}
	m.UnstartedTransactions = m.UnstartedTransactions[:0]
	return
}
```